packages/@aws-cdk/aws-stepfunctions-tasks/lib/invoke-function.ts

```diff
@@ -2,26 +2,53 @@ import iam = require('@aws-cdk/aws-iam');
 import lambda = require('@aws-cdk/aws-lambda');
 import sfn = require('@aws-cdk/aws-stepfunctions');
 
+/**
+ * Properties for InvokeFunction
+ */
+export interface InvokeFunctionProps {
+  /**
+   * The JSON that you want to provide to your Lambda function as input.
+   */
+  readonly payload?: { [key: string]: string };
+
+  /**
+   * Whether to pause the workflow until a task token is returned
+   *
+   * @default false
+   */
+  readonly waitForTaskToken?: boolean;
+}
+
 /**
  * A StepFunctions Task to invoke a Lambda function.
  *
  * A Function can be used directly as a Resource, but this class mirrors
  * integration with other AWS services via a specific class instance.
  */
 export class InvokeFunction implements sfn.IStepFunctionsTask {
-  constructor(private readonly lambdaFunction: lambda.IFunction) {
+
+  private readonly waitForTaskToken: boolean;
+
+  constructor(private readonly lambdaFunction: lambda.IFunction, private readonly props: InvokeFunctionProps) {
+    this.waitForTaskToken = props.waitForTaskToken === true;
   }
 
   public bind(_task: sfn.Task): sfn.StepFunctionsTaskProperties {
     return {
-      resourceArn: this.lambdaFunction.functionArn,
+      resourceArn: this.waitForTaskToken
```

Comment:
The case where "Resource" equals to "**arn:aws:states:::lambda:invoke**" (let's call it magic ARN) is missing. It is shown in the first code snippet in the [documentation](https://docs.aws.amazon.com/step-functions/latest/dg/connect-lambda.html) and is not equivalent to the case where "Resource" is lambda function ARN.

When we define the function ARN as "Resource", the input of your lambda function can be overridden by "Parameters", if the field is present. If not, the state input will be passed to Lambda. However, when we use the magic ARN or ".waitForTaskToken", the input can be only obtained by "Payload", the subfield of "Parameters".

That's why we can't neglect the case where "Resource" is "**arn:aws:states:::lambda:invoke**". This particular case exists only for Lambda because Step Functions used to invoke Lambda in the same way as "Activity".

My advice is to implement magic ARN to CDK to guarantee the complete service integration for Lambda. We can add a new variable in the interface "InvokeFunctionProps" to let users decide if they want to invoke Lambda as integrated service or not. When users prefer to invoke Lambda in the legacy way (by function ARN), we should validate that the variable "waitForTaskToken" is not true.